pkg/orchestrator/etcd_worker.go

```diff
@@ -319,3 +374,25 @@ func (worker *EtcdWorker) cleanUp() {
 	worker.revision = 0
 	worker.pendingUpdates = worker.pendingUpdates[:0]
 }
+
+func (worker *EtcdWorker) isDeleteCounterKey(key []byte) bool {
+	return string(key) == worker.prefix.String()+deletionCounterKey
+}
+
+func (worker *EtcdWorker) handleDeleteCounter(value []byte) {
+	if len(value) == 0 {
+		// The delete counter key has been deleted, resetting the internal counter
+		worker.deleteCounter = 0
+		return
+	}
+
+	var err error
+	worker.deleteCounter, err = strconv.ParseInt(string(value), 10, 64)
+	if worker.deleteCounter <= 0 {
+		log.Panic("unexpected delete counter", zap.Int64("value", worker.deleteCounter))
+	}
+	if err != nil {
+		// This should never happen unless Etcd server has been tampered with.
+		log.Panic("strconv failed. Unexpected Etcd state.", zap.Error(err))
+	}
```

Comment:
Better to check `err != nil` first, when err is not nil, `worker.deleteCounter` is 0